scmail.py

```diff
@@ -134,42 +143,61 @@ def register(ctx, fingerprint):
 
 
 @click.command()
-@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox', required=True, help='The fingerprint of the yourself mailbox.')
-@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender', default=None, help='The senders fingerprint.')
-@click.password_option('--password', '-p', prompt='Enter password of private key', help='The passphrase of private key')
+@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox',
+              required=True, type=str, help='The fingerprint of the yourself mailbox.')
+@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender',
+              type=str, help='The senders fingerprint.')
+@click.password_option('--password', '-p', prompt='Enter password of private key',
+                       help='The passphrase of private key')
 @click.pass_context
 def retrieve(ctx, fingerprint, sender_fingerprint, password):
     """Retrieve and Post messages from API"""
+    # Prepare send data.
     logging.debug(fingerprint, sender_fingerprint)
     data = {'fingerprint': fingerprint}
-    if not sender_fingerprint:
+    if sender_fingerprint == '':
```

Comment:
It looks like this is checking to see if a `sender_fingerprint` is empty and only adding it to the request payload if there isn't one. Because it's not required the check can be dropped and `data` can be initialized with it, regardless of whether it has a value or not. Unless I'm overlooking something here?

```python
data = {
    "fingerprint": fingerprint,
    "sender_fingerprint": sender_fingerprint,
}
```

Reply:
The develop branch has a fix for this. This branch needs to be caught up with develop.

Reply:
I remember that now.